### ISSUE TITLE
Failed to load plugin: Error: Cannot find module 'members-area/node_modules/entities'

### ISSUE THREAD

**Issue by jlowe64** [members-area/members-area#43]
Hi, 

I'm trying to get this up and running to see if our space could use it. I'm getting this error. I've looked in the folder structure and it has entitites installed. I've also installed it globally just incase it needed that, but it doesn't seem to have helped at all. 

Any suggestions?

Thanks in advance.

J

**Comment by jlowe64** [comment]
I seem to have gotten it to work by using the bootstrap.sh file, however I'm still getting issues with sqlite bindings apparnetly not being installed. Can't seem to find an appropriate repo for it.

**Comment by benjie** [comment]
Hi @jlowe64!

First, a warning: the bootstrap is only intended for development - don't use it on production. I've also not published to npm in a very long time (mostly because I didn't think anyone was using it so didn't bother) so you'll want to use the version from git directly.

TL;DR: oddly I've found that running `npm install sqlite3@; npm install sqlite3@` solves my issues - it seems to need to be called twice.

The members area itself is intended to be used as a dependency (rather than using it directly); so here's an example of something that uses it:

https://github.com/somakeit/somakeit-members-area

I too was having issues with the SQLite library; in production we're still running the Members Area on Node 0.12 since I've not had time to update all its dependencies; however I've had some luck running it locally with Node 4/Node 5 but as you say, I had to sort out the SQLite bindings and probably a few other things too - simply with `npm install sqlite3@` to install the latest version of the sqlite library.

Here's an `npm shrinkwrap` of my members_area folder I use in development:

https://gist.github.com/benjie/541f4276ac3f1589194b

When developing after using `npm install` in the `somakeit-members-area` folder; I then go into `node_modules`, remove the `members-area` folder, and then `git clone git@github.com:members-area/members-area members-area` so I have a literal checkout of the members-area inside node_modules. The reason for this is quite a few of the plugins I recently merged into the members-area core do things like `require 'members-area/app/controllers/role'` - and for that to work `members-area` needs to be directly in a `node_modules` folder in the Node path for the plugin file, which doesn't work if you use the traditional `npm link` mechanic.

**Comment by benjie** [comment]
Hi again @jlowe64; ignore all that rubbish - I bit the bullet and just fixed it and published a new release to npm.

Members-area 0.5.0 is now on npm and it runs under Node 5.1.1 and supports the `members quickstart` as shown in the README. I've done it a few times into a clean folder to ensure it works: 

`npm install -g members-area && mkdir myarea && cd myarea && members quickstart`

Apologies for any time wasted over this!

**Comment by jlowe64** [comment]
@benjie you honestly do not have to apologize for anything. I have a space up in Wembley area that I'd like to use this for. I was actually midstream in developing my own version of this exact thing. I was researching gocardless node integrations and magically came across your code here. It's almost exactly what we will be needing.

Thank you for all the hard work. I may have lots of dumb questions getting this going, so be prepared. You truly are amazing for doing this package.

**Comment by benjie** [comment]
Thanks @jlowe64 ; Pull Requests always welcome!

The GoCardless integration is pretty solid in the members area; we've never had any issues. Banking on the other hand is somewhat complex because Barclays recently (about 9 months ago) started changing the `FITID` field for some previous transactions; meaning we were seeing them as multiple different transactions even though they were not. Turns out that we can't use the description/amount/date combination as a unique reference either. Our solution (only implemented a few days ago) is to extract the specific payment reference (user ID, e.g. `M000137`) from the description and use that along with the date and amount to uniquely identify transactions. It's a serious kludge; but I'm not sure what else we can do given that Barclays no longer respect the uniqueness and immutability of the FITID.

Could I perhaps request you use our referral code for GoCardless if it's not already too late? Gets us both £50 off our fees:

https://gocardless.com/?r=Z65JVARP&utm_source=website&utm_medium=copy_paste&utm_campaign=referral_scheme_50

There's one glaring omission in the members area currently - and that's that I apparently never got around to writing the forgot password functionality. It's on my lengthy todo list...

**Comment by benjie** [comment]
Closing this issue because I believe it's now solved; please re-open if it doesn't work for you.

**Comment by jlowe64** [comment]
Sorry, I'm running into some other issues:

I got the server up and running when I did "members quickstart" however when I just do "members run" it seems to throw sqlite database errors. I've tried to see what the difference is between quickstart and run is, but I think it's whatever's been done in the init script. It doesn't seem to be loading environment variables. 

Any help appreciated!

**Comment by benjie** [comment]
Weird; try `npm start` instead of `members run`

**Comment by benjie** [comment]
Also ensure that any old versions and your node_modules folder have been completely expunged; it could be that it's trying to run the sqlite3 2.2 version instead of 3.1.1 - ensure there's no other `node_modules` folders in the path of your members area; for example, if it's `/a/b/c/d/myarea` ensure that there's no `/a/b/c/d/node_modules` or `/a/b/c/node_modules` or `/a/b/node_modules` and so on
